### PR TITLE
docs: Adding testing plan document, to introduce Testing Strategy

### PR DESCRIPTION
**Pull request by ngrkajac**
**Description**

Changes proposed in this pull request:

- Adding a Testing Plan document as a base to create and design a Testing Strategy

### CODE REVIEW

File: docs/contributor/testing-plan.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+fter implementing all necessary corrective actions, it is important to conduct a regular audit to ensure that the agreed-upon practices, policies, and objectives continue to be followed. The following steps should be taken:
+
+- [ ] Conduct the regular audit.
+- [ ] Reflect on the results of the audit and determine if any additional corrective actions are needed.
+- [ ] Brainstorm on how to add more quality attributes and activities.
+
+### Schedule
+
+**TBD.** To define this section, we need to agree on the proposal outlined in this document, assess our development budget, and devise feasible timelines for each phase, breaking it down into smaller steps. A Gantt diagram with milestones and timelines will also be needed.
```
Comment (IwonaLanger) [inline docs/contributor/testing-plan.md:91]:
```suggestion
**TBD.** To define this section, we must agree on the proposal outlined in this document, assess our development budget, and devise feasible timelines for each phase, breaking it down into smaller steps. A Gantt diagram with milestones and timelines will also be needed.
```

File: docs/contributor/testing-plan.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+| Aspect               | Document                            | Purpose                                                      |
+| -------------------- | ----------------------------------- | ------------------------------------------------------------ |
+| Software Engineering | Code Style Code of Conduct          | This document defines the coding baseline for the software engineering activities. |
+| Software Engineering | Code Review Code of Conduct         | This document outlines the baseline expectations for the code-review process, including which test suites are mandatory and which are optional, the necessity of system diagrams for architectural changes, and more... |
+| Software Engineering | Project Structure                   | This document includes the agreement on the architecture and structure of the project. |
+| Monitoring           | The metrics generated by the module | This document outlines the observation-critical metrics to be used for monitoring the module. |
+| Integration Testing  | External dependencies isolation     | This document outlines the approach to replace third-party dependencies with controllable test-doubles. |
```
Comment (IwonaLanger) [inline docs/contributor/testing-plan.md:133]: If these are existing docs, please provide the links.

File: docs/contributor/testing-plan.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
+# Cloud Manager Testing Strategy Introduction Plan
+
+### Purpose
+
+The Cloud Manager component has reached maturity and is approaching production readiness. This means that ensuring quality is critical for an upcoming successful launch, extension, maintenance, and operation.
```
[the remark this answers is not in the document]
Comment (ngrkajac) [reply docs/contributor/testing-plan.md:5]: Iwona, this is a great question overall. Cloud Manager is a mix of modules and components. It is a component in the Central Plane but acts as a module in how a user can enable or disable it.

Although a more accurate identifier would be a component, you are probably right - in order not to confuse customers, let me rename it to the module.

File: docs/contributor/testing-plan.md (hunk 2 above)
Comment (ngrkajac) [reply docs/contributor/testing-plan.md:133]: The documents don't exist or are not yet adequately defined.

File: docs/contributor/testing-plan.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+| Artefact                     | Links                                                        |
+| ---------------------------- | ------------------------------------------------------------ |
+| Unit test coverage           | ● Unit test coverage expectation document<br />● Unit test coverage report |
+| Functional correctness       | ● The functional/acceptance testing report<br />● Functional testing approach outline |
+| Functional completeness      | ● Coverage analysis of functional aspects by the acceptance test suite |
+| Performance                  | ● Module expected Service Level Agreements (SLAs) <br />● Performance testing strategy and approach<br />● Performance testing report which includes time behaviour and resource utilization |
+| Maintainability & Modularity | ● Code review code of conduct <br />● Code-style agreement<br />● Project structure description<br />● Modular metrics and expectations |
+| Reusability                  | ● TBD. How reusable is the codebase?                         |
+| Testability                  | ● TBD. How Testable os the codebase?                         |
```
Comment (grego952) [inline docs/contributor/testing-plan.md:118]:
```suggestion
| Testability                  | ● TBD. How testable is the codebase?                         |
```

**Review by IwonaLanger — APPROVED** [review]

**Review by dushanpantic — APPROVED** [review]

**Pull request merged**